src/extract-functions.js

```diff
@@ -1,11 +1,17 @@
 import isFunction from './isFunction';
 
-export default function (...args) {
-  let result = [];
-  for (let i = 0; i < args.length; i += 1) {
-    const arg = args[i];
-    if (isFunction(arg)) result.push(arg);
-    else if (Array.isArray(arg)) result = result.concat(arg.filter(isFunction));
+export default function extractFunctions() {
+  const fns = [];
+
+  for (let i = 0; i < arguments.length; i += 1) {
+    const arg = arguments[i];
+
+    if (isFunction(arg)) {
+      fns.push(arg);
+    } else if (Array.isArray(arg)) {
+      fns.push(...(extractFunctions(...arg) || []));
```

Comment:
Perhaps instead of pushing empty array it might be better to actually check its length before doing so. It's probably micro optimization and could use some measurement. Also it might be more readable. I had to read this line 3 times to actually understand what's happening there.